students/mmancini/lesson07/parallel.py

```diff
@@ -0,0 +1,313 @@
+                continue
+            elif product['quantity_available'] == '':
+                continue
+            elif product['quantity_available'] == 'quantity_available':
+                continue
+            else:
+                py_dict[product['product_id']] = dict([('description',
+                                                        product['description']),
+                                                       ('product_type',
+                                                        product['product_type']),
+                                                       ('quantity_available',
+                                                        product['quantity_available'])])
+        return py_dict
+
+def show_rentals(product_id):
+    '''
+        des:
+            show customer products rented for given product id
+        in:
+            product id
+        out:
+            dictionary of customers who rented given product
+    '''
+    mongo = MongoDBConnection()
+    with mongo:
+        database = mongo.connection.hp_norton
+        customer_collection = database["Customers"]
+        rental_collection = database["Rentals"]
+        py_dict = {}
+        for renter in rental_collection.find():
+            if product_id == renter['product_id']:
+                for customer in customer_collection.find():
+                    if renter['user_id'] == customer['user_id']:
+                        py_dict[customer['user_id']] = dict([('name',
+                                                              customer['name']),
+                                                             ('address',
+                                                              customer['address']),
+                                                             ('phone_number',
+                                                              customer['phone_number']),
+                                                             ('email',
+                                                              customer['email'])])
+    return py_dict
+
+def dbs_cleanup():
+    '''drop the db'''
+    mongo = MongoDBConnection()
+    with mongo:
+        database = mongo.connection.hp_norton
+        database["Customers"].drop()
+        database["Products"].drop()
+        database["Rentals"].drop()
+    return 'databases dropped'
+
+
+def main():
+
+    '''
+        process import data in parallel for products and customers.
+    '''
+
+    dbs_cleanup()
+
+    start_time = time.time()
+
+    path = ('csv_files')
+    thread1 = threading.Thread(target=import_products,
+                               args=(path, 'product_file.csv'), daemon=True)
+    thread2 = threading.Thread(target=import_customers,
+                               args=(path, 'customer_file.csv'), daemon=True)
+    thread1.start()
+    thread2.start()
+
+    thread1.join()
+    thread2.join()
+
+    end_time = time.time()
+
+    thread1_result = output_queue.get()
```

Comment:
This is only true if _thread1_ is the first to push data into _output_queue_. In the (very possible) scenario that _thread2_ puts its data into the queue first, _thread1_result_ will really contain the results from _thread2_.

Try to add a marker or something else that allows you to identify whose data you're pulling from the queue.